File: VB/TreeMapHierarchicalDataAdapterSample/DataModel.vb
Imports System.Collections.Generic

Namespace TreeMapHierarchicalDataAdapterSample

    Public Class Country

        Public Property Name As String

        Public Property Gdp As Double
    End Class

    Public Class SideOfWorld

        Public Property Name As String

        Private countriesField As List(Of Country) = New List(Of Country)()

        Public ReadOnly Property Countries As List(Of Country)
            Get
                Return countriesField
            End Get
        End Property
    End Class
End Namespace

File: CS/TreeMapHierarchicalDataAdapterSample/DataModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TreeMapHierarchicalDataAdapterSample {
    public class Country {
        public string Name { get; set; }
        public double Gdp { get; set; }
    }

    public class SideOfWorld {
        public string Name { get; set; }

        List<Country> countries = new List<Country>();
        public List<Country> Countries { get { return countries; } }
    }
}
